plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -3760,7 +3760,7 @@ public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session,
                 name.getCatalogName(),
                 new SchemaTableName(
                         name.getSchemaTableName().getSchemaName(),
-                        name.getSchemaTableName().getTableName() + tableNameSplit.getSuffix().orElse(""))));
+                        name.getSchemaTableName().getTableName() + tableNameSplit.suffix().orElse(""))));
```

Comment:
use `tableName.getTableName()` and remove `TableNameSplitResult` class
see
https://github.com/trinodb/trino/blob/f5084beedef43eca07d4257bff71ec7217e921b6/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java#L656-L663